assets/hip-206/solidity/IHederaTokenService.sol

```diff
@@ -49,6 +49,148 @@ interface IHederaTokenService {
+        uint32 numerator;
+
+        // A rational number's denominator, used to set the amount of a value transfer to collect as a custom fee
+        uint32 denominator;
+
+        // The minimum amount to assess
+        uint32 minimumAmount;
+        
+        // The maximum amount to assess (zero implies no maximum)
+        uint32 maximumAmount;
+        bool netOfTransfers;
+
+        // The ID of the account to receive the custom fee, expressed as a solidity address
+        address feeCollector;
+    }
+
+    /// A fee to assess during a transfer that changes ownership of an NFT. Defines the fraction of
+    /// the fungible value exchanged for an NFT that the ledger should collect as a royalty. ("Fungible
+    /// value" includes both ℏ and units of fungible HTS tokens.) When the NFT sender does not receive
+    /// any fungible value, the ledger will assess the fallback fee, if present, to the new NFT owner.
+    /// Royalty fees can only be added to tokens of type type NON_FUNGIBLE_UNIQUE.
+    struct RoyaltyFee {
+        // A fraction's numerator of fungible value exchanged for an NFT to collect as royalty
+        uint32 numerator;
+
+        // A fraction's denominator of fungible value exchanged for an NFT to collect as royalty
+        uint32 denominator;
+
+        // If present, the fixed fee to assess to the NFT receiver when no fungible value is exchanged
+        // with the sender
+        FixedFee fixedFee;
+
+        // The ID of the account to receive the custom fee, expressed as a solidity address
+        address feeCollector;
+    }
+
+    /// A Key can be a public key from either the Ed25519 or ECDSA(secp256k1) signature schemes, where
+    /// in the ECDSA(secp256k1) case we require the 33-byte compressed form of the public key. We call
+    /// these public keys <b>primitive keys</b>.
+    /// A Key can also be the ID of a smart contract instance, which is then authorized to perform any 
+    /// precompiled contract action that requires this key to sign. 
+    /// Note that when a Key is a smart contract ID, it <i>doesn't</i> mean the contract with that ID 
+    /// will actually create a cryptographic signature. It only means that when the contract calls a 
+    /// precompiled contract, the resulting "child transaction" will be authorized to perform any action
+    /// controlled by the Key.
+    /// Exactly one of the possible values should be populated in order for the Key to be valid.
+    struct Key {
+        // bit field representing the key type. Keys of all types that have corresponding bits set to 1
+        // will be created for the token.
+        // 0th bit: adminKey
+        // 1st bit: kycKey
+        // 2nd bit: freezeKey
+        // 3rd bit: wipeKey
+        // 4th bit: supplyKey
+        // 5th bit: feeScheduleKey
+        // 6th bit: pauseKey
+        // 7th bit: ignored
+        uint8 keyType;
```

Comment:
Unless this is a "packed" struct we should kale it at least uint16.  since it's not intended to "go over the wire" there is no need to pack it. perhaps just `uint`?